packages/coil-extension/src/background/services/AuthService.ts

```diff
@@ -120,108 +152,19 @@ export class AuthService extends EventEmitter {
     return this._op
   }
 
-  tokenInvalid(token: string | null) {
-    return !token || tokenUtils.isExpired({ token })
+  initialize() {
+    SuperTokens.init({
```

Comment:
What is this actually doing ? 

If it's only initiated so the signOut call works, then perhaps it's not even needed? 
Cause the log out will be done on coil.com, no? 
Currently, the extension listens for `coil_writeToken` and when it's an empty string, a logout is inferred. 
It then clears its own copy of the token, so that it doesn't go about injecting it straight back in.
With cookies, there's no real auth state maintained and the store just needs to be cleared of the cached user.

Reply:
Is this what keeps the cookie fresh when you don't go to coil.com ?

Reply:
https://supertokens.com/docs/website/index.html#sending-requests-with-fetch

So window.fetch is wrapped and monkey patched?

Reply:
So with that wrapping, we shouldn't even really need to add the `credentials:include` 
What does it actually do when it wraps the function?
SuperTokens supports bearer tokens as well as cookies right? 
So the wrapping would include those ? 

In any case (if we haven't already) we should probably put the SuperTokens import inside background.ts, right at the top, before other imports. 

> we may have already included this in our /signout path

As it stands, the extension doesn't have a dedicated logout UI.  It just tracks the state of coil.com, listening to logout events (coil_writeToken with an empty string)

How will the extension know a logout has happened now? Is there a logout event emitted?

Reply:
You're right about the credentials, looks like unless you explicitly say otherwise, it'll [automatically include them](https://github.com/supertokens/supertokens-website/blob/10285c14dd8a47b72098175f7f473adf3ffaf8cc/lib/ts/utils/index.ts#L99-L102).

With SuperTokens, it's a little funky but the access token cookie by default is a JWT-like token (but it's missing a few key props [unless modified](https://supertokens.com/docs/thirdpartyemailpassword/common-customizations/sessions/with-jwt/about)). It is possible to [store this token in localStorage](https://supertokens.com/docs/thirdpartyemailpassword/advanced-customizations/examples/localstorage/about#frontend), but it looks like you have to modify the frontend manually.

> How will the extension know a logout has happened now? Is there a logout event emitted?

If you were to log out from another window, it wouldn't be immediately apparent to the extension until the next time it attempted an API request (reading whoami data, fetching an anonymous token, tipping, etc...) in which case the browser would not have any access/refresh token cookies to pass along and the request would fail.

Reply:
> If you were to log out from another window, it wouldn't be immediately apparent to the extension
Can we add an event then?
At present, when we logout, we update the browser action icon immediately

Reply:
Does it do that by constantly polling localStorage? And if so, does it decrypt the JWT each time to know when the token is expired as well?

Reply:
No, we bind to the `coil_writeToken` event from in the content script ( extension component that runs within a page ) when it's running under coil.com. 

We check once on startup to catch logouts that happened while the extension was temporarily disabled.

Reply:
I just checked and we're still dispatching that event on the signout page (I'm putting in a change to make sure that occurs AFTER we perform the SuperTokens signout).

I can't change the name of the event because of the current versions of the extension. Likewise, we should be sending the event on signin. It looks like it is but I'll double-check with Mike tomorrow to make sure.

Right now we're still setting the legacy JWT so this should all still work, but in the long run, I guess we should update the content script's `syncViaInjectToken`/`injectToken` communication with the background script to do something like check the user via the API, or use some non-token localStorage flag whether the user is authed.

Reply:
Would be good to have a comment on what the init call does (wraps fetch, refreshes tokens automatically etc)